src/main/java/school/hei/haapi/endpoint/rest/controller/StudentController.java

```diff
@@ -41,19 +43,21 @@ public Student getStudentById(
   @GetMapping("/students")
   public List<Student> getStudents(
       @RequestParam PageFromOne page, @RequestParam("page_size") BoundedPageSize pageSize) {
-    return userService
-        .getByRole(User.Role.STUDENT, page, pageSize).stream()
-        .map(userMapper::toRestStudent)
-        .collect(toUnmodifiableList());
+    List<Student> students =  userService.getByRole(User.Role.STUDENT, page, pageSize).stream()
+            .map(userMapper::toRestStudent)
+            .collect(toUnmodifiableList());
+    return students;
   }
 
   @PutMapping("/students")
-  public List<Student> saveAll(@RequestBody List<Student> toSave) {
+  public List<Student> saveAll(@RequestBody List<Student> toWrite) {
+    if (toWrite.size() > EventConf.MAX_EVENT_REQUEST_ENTRY) {
```

Comment:
Now that I think about it, the check should probably be performed at the service level instead of the controller level. The main advantage being that we will be sure any call to `UserService.saveAll` would be checked before sending event.

Reply:
Now that I think even more about it (yes, I do think a lot), the checking whether a payload to produce over the event bus is valid MUST be a responsibility of `EventProducer`, as the name suggests.

Hence you should not leak `EventConf.MAX_EVENT_REQUEST_ENTRY` publicly. You should create a **public** method `boolean EventProducer.isPayloadValid(List<T>)` which will do its calculation using a **private** `static final in EventProducer.MAX_PUT_EVENT_ENTRIES`.

Then in `UserService` you would just call `EventProducer.isPayloadValid` before calling `EventProducer.sendEvent`.